### PR TITLE
[SPARK-38014][SQL][TESTS] Add Parquet Data Page V2 write scenario for `BuiltInDataSourceWriteBenchmark`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
`BuiltInDataSourceWriteBenchmark` only test data page V1  for parquet now, this pr add parquet data page V2 write scenario and update relevant benchmark result files.

### Why are the changes needed?
Add micro benchmark scene for parquet

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GA

### CODE REVIEW

**Comment by LuciferYang** [comment]
Wait GA generate results

**Comment by LuciferYang** [comment]
result format changed, need re-generate

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/BuiltInDataSourceWriteBenchmark.scala [hunk 1]
```diff
@@ -53,7 +56,16 @@ object BuiltInDataSourceWriteBenchmark extends DataSourceWriteBenchmark {
 
     formats.foreach { format =>
       runBenchmark(s"$format writer benchmark") {
-        runDataSourceBenchmark(format)
+        if (format.equals("Parquet")) {
+          ParquetProperties.WriterVersion.values().zip(Seq("V1", "V2")).foreach {
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/BuiltInDataSourceWriteBenchmark.scala:60]: In general, we had better avoid magic string. Please use `ParquetProperties.WriterVersion.PARQUET_1_0.toString` and `ParquetProperties.WriterVersion.PARQUET_2_0.toString`.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you for making this, @LuciferYang . In general, looks good with one minor comment.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/BuiltInDataSourceWriteBenchmark.scala (hunk 1 above)
Comment (LuciferYang) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/BuiltInDataSourceWriteBenchmark.scala:60]: ok, I will fix this later

Comment (LuciferYang) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/BuiltInDataSourceWriteBenchmark.scala]: 5155fd7 fix this, let me update the bench-results

Comment (LuciferYang) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/BuiltInDataSourceWriteBenchmark.scala]: @dongjoon-hyun done

**Review by dongjoon-hyun — APPROVED** [review]
Thank you for updates. +1, LGTM, @LuciferYang .
Merged to master.

**Pull request closed without merging**